catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/ModalRegistry.view.js

```diff
@@ -15,264 +15,292 @@
-                                wreqr.vent.trigger('refreshRegistries');
-                                view.closeAndUnbind();
-                            });
-                        } else {
+                this.$el.attr('role', "dialog");
+                this.$el.attr('aria-hidden', "true");
+                this.$el.attr('data-backdrop', "static");
+                this.renderDetails(config);
+                this.initRadioButtonsUI(properties);
+                if (this.mode !== 'edit') {
+                    this.showError('Registry must have a valid url (non-blank/non-duplicate)');
+                }
+                this.rebind(properties);
+            },
+            submitData: function () {
+                wreqr.vent.trigger('beforesave');
+                var view = this;
+                var service = this.getConfig();
+                if (service) {
+                    service.save().then(function () {
                             wreqr.vent.trigger('refreshRegistries');
                             view.closeAndUnbind();
-                        }
-                    },
-                    function () {
-                        wreqr.vent.trigger('refreshRegistries');
-                    }).always(function () {
+                        },
+                        function () {
+                            wreqr.vent.trigger('refreshRegistries');
+                        }).always(function () {
                         view.closeAndUnbind();
                     });
-            }
-        },
-        onClose: function() {
-            this.modelBinder.unbind();
-            this.$el.off('hidden.bs.modal');
-            this.$el.off('shown.bs.modal');
-        },
-        closeAndUnbind: function() {
-            this.modelBinder.unbind();
-            this.$el.modal("hide");
-        },
-        rebind: function (properties) {
-            var $boundData = this.$el.find('.bound-controls');
-            var bindings = Backbone.ModelBinder.createDefaultBindings($boundData, 'name');
-            delete bindings.value;
-            this.modelBinder.bind(properties, $boundData, bindings);
-        },
-        initRadioButtonsUI: function(boundModel) {
-            var $radios = this.$el.find('input[type=radio]');
-            var view = this;
-
-            _.each($radios, function(radio) {
-                var $radio = view.$(radio);
-                var $label = $radio.closest('label.btn');
+                }
+            },
+            urlChanged: function (evt) {
+                var url = evt.target.value;
+                var hostname;
+                if (url) {
+                    hostname = this.extractHostnameFromUrl(url);
+                }
+                if (hostname && this.urlIsValid(url)) {
+                    var oldName;
+                    var nameToSet = hostname;
+                    var toCheck = "(" + hostname + ")";
+                    if (!this.type || this.type === "CSW Registry Store") {
```

Comment:
Think it would be cleaner to just use the `type` here instead of converting one particular type to a special value.